src/wrappers/cryptosignwrapper.cpp

```diff
@@ -64,6 +79,59 @@ bool verify(std::string const &signature, std::string const &message, bytes cons
     return res == 0;
 }
 
+bytes scrypt(std::string const &password, KdfParams const &kdfParams)
+{
+    unsigned int const keyLength = kdfParams.dklen;
+    unsigned char derivedKey[keyLength];
+
+    auto passw = reinterpret_cast<const unsigned char*>(password.data());
+    auto salt = reinterpret_cast<const unsigned char*>(kdfParams.salt.data());
+
+    if (crypto_pwhash_scryptsalsa208sha256_ll
+            (passw, password.size(),
+             salt, kdfParams.salt.size(),
+             kdfParams.n,
+             kdfParams.r,
+             kdfParams.p,
+             derivedKey, keyLength) !=0)
+    {
+        throw std::runtime_error(ERROR_MSG_SCRYPTSY);
+    }
+
+    return bytes(derivedKey, derivedKey + keyLength);
+}
+
+std::string hmacsha256(bytes const &key, std::string const &cipherText)
+{
+    auto k = reinterpret_cast<const unsigned char*>(key.data());
+    auto cipher = reinterpret_cast<const unsigned char*>(cipherText.data());
+
+    unsigned char digest[HMAC_SHA256_BYTES];
+
+    crypto_auth_hmacsha256_state state;
+
+    crypto_auth_hmacsha256_init(&state, k, key.size());
+    crypto_auth_hmacsha256_update(&state, cipher, cipherText.size());
+    crypto_auth_hmacsha256_final(&state, digest);
+
+    return std::string(digest, digest + HMAC_SHA256_BYTES);
+}
+
+bytes aes128ctrDecrypt(bytes const &key, std::string cipherText, std::string const &iv)
```

Comment:
Can we perhaps write some "direct" unit tests for this function - and the one above (that is, not only via `deriveSecretKey`)? 

(perhaps not worth the effort, though, it's just parameter forwarding)

Reply:
Hmm, would be nice to have some direct tests, however I also don't think it might be worth the effort, since it is indeed just a parameter forwarding